gillespy2/core/model.py

```diff
@@ -857,28 +857,34 @@ def get_best_solver(self):
         """
         from gillespy2.solvers.numpy import can_use_numpy
         hybrid_check = False
-        if len(self.get_all_assignment_rules()) or len(self.get_all_rate_rules())  \
-                or len(self.get_all_function_definitions()) or len(self.get_all_events()):
+        chybrid_check = True
+        if len(self.get_all_rate_rules())  or len(self.get_all_events()):
             hybrid_check = True
+        if len(self.get_all_assignment_rules()) or len(self.get_all_function_definitions()):
+            hybrid_check = True
+            chybrid_check = False
 
         if len(self.get_all_species()) and hybrid_check == False:
             for i in self.get_all_species():
                 tempMode = self.get_species(i).mode
                 if tempMode == 'dynamic' or tempMode == 'continuous':
                     hybrid_check = True
                     break
-        if can_use_numpy and hybrid_check:
+
+        from gillespy2.solvers.cpp.build.build_engine import BuildEngine
+        can_use_cpp = not len(BuildEngine.get_missing_dependencies())
+
+        if can_use_cpp and hybrid_check and chybrid_check:
+            from gillespy2 import TauHybridCSolver
+            return TauHybridCSolver
+        elif can_use_numpy and hybrid_check:
             from gillespy2 import TauHybridSolver
             return TauHybridSolver
-
         elif not can_use_numpy and hybrid_check:
             raise ModelError('TauHybridSolver is the only solver currently that supports '
                              'AssignmentRules, RateRules, FunctionDefinitions, or Events. '
                              'Please install Numpy.')
```

Comment:
'Please install Numpy.'  This is probably not the right solution.  Needs to be revised